### PR TITLE
fix for {12,0} {am,pm} cases

### LINKED ISSUE #25: format unformat of {12,0} {am,pm},  24
I was looking at `adjustMeridiem` function it seemed a bit strange and then i took look at moment.js behaviour:
```js
moment("0 AM", "h a").format() // "2017-07-14T00:00:00+04:00"
moment("0 PM", "h a").format() // "2017-07-14T12:00:00+04:00"
moment("12 PM", "h a").format() // "2017-07-14T12:00:00+04:00"
moment("12 AM", "h a").format() // "2017-07-14T00:00:00+04:00"

moment("00", "hh").format() // "2017-07-14T00:00:00+04:00"
moment("12", "hh").format() // "2017-07-14T12:00:00+04:00"
moment("24", "hh").format() // "2017-07-15T00:00:00+04:00"

moment("00", "hh").format("h a") // "12 am"
moment("12", "hh").format("h a") // "12 pm"
moment("24", "hh").format("h a") // "12 am"
```
And our implementation clearly does not match it.

Having range like `validateRange 0 59` for minute and seconds matches moment behaviour:
```js
moment("60", "mm").format() // "Invalid date"
moment("60", "ss").format() // "Invalid date"
```

But as we use  `validateRange 0 23` and `validateRange 0 11` for hour we behave in different way (i.e. can't parse input containing 24 or 12). is this what we want?

If we want to match moment then we should change `adjustMeridiem` to something like this:
```purescript
adjustMeridiem ∷ Int → Meridiem → Int
adjustMeridiem 12 AM = 0
adjustMeridiem 12 PM = 12
adjustMeridiem n  PM = n + 12
adjustMeridiem n  AM = n
```

and change `validateRanges` to include 24 and 12

### PR DESCRIPTION
**Pull request by safareli**
fix #25

### CODE REVIEW

**Comment by safareli** [comment]
it need to fail on: `24:1:40`

File: test/src/DateTime.purs [hunk 1]
```diff
@@ -48,6 +50,24 @@ datetimeTest = describe "Data.Formatter.DateTime" do
       (void $ format `FDT.unformatDateTime` dateStr) `shouldEqual` (Right unit)
     )
 
+  describe "hour 24" do
+    it "24 hour and more then 0 time component" $ do
+      let err = (Left "When hour is 24, other time component must be 0@1:24")
+      shouldEqual (FDT.unformatDateTime "YYYY-DD-MM HH:mm:ss:SSS" "0000-01-01 24:00:00:001") err
+      shouldEqual (FDT.unformatDateTime "YYYY-DD-MM HH:mm:ss:SSS" "0000-01-01 24:00:01:000") err
+      shouldEqual (FDT.unformatDateTime "YYYY-DD-MM HH:mm:ss:SSS" "0000-01-01 24:01:00:000") err
```
Comment (garyb) [inline test/src/DateTime.purs]: How do these pass? Is the `"@1:24"` in the error some magic?

Comment (safareli) [inline test/src/DateTime.purs]: it's represents position https://github.com/slamdata/purescript-formatters/blob/master/src/Data/Formatter/Parser/Utils.purs#L24

Comment (garyb) [inline test/src/DateTime.purs]: Can we print it as "When hour is 24, other time component must be 0 (line 1, col 24)" or something like that instead?

Comment (safareli) [inline test/src/DateTime.purs]: sure

**Pull request merged**